### PR TITLE
[CI][SYCL-CTS] Small improvements

### PR DESCRIPTION
**Pull request by KornevNikita**
Remove step "List excluded SYCL CTS categories" due to duplicated code.
List cts-filter in the "Build SYCL CTS tests" step instead.
Also remove DSYCL_CTS_MEASURE_BUILD_TIMES. It affects the performance, but we don't use these results.

### CODE REVIEW

**Comment by KornevNikita** [comment]
@aelovikov-intel is it what you were [talking](https://github.com/intel/llvm/pull/14325#discussion_r1691818740) about?

File: .github/workflows/sycl-linux-run-tests.yml [hunk 1]
```diff
@@ -303,24 +303,24 @@ jobs:
...
       run: |
-        # Do not exclude any category if passing the CTS_TESTS_TO_BUILD variable.
         cts_exclude_filter=""
+        # If CTS_TESTS_TO_BUILD is null - use filter
         if [ -z "$CTS_TESTS_TO_BUILD" ]; then
           if [ "${{ contains(inputs.target_devices, 'opencl:cpu')  }}" = "true" ]; then
             cts_exclude_filter=$PWD/devops/cts_exclude_filter_OCL_CPU
-          fi
-          if [ "${{ contains(inputs.target_devices, 'level_zero:gpu')  }}" = "true" ]; then
+          elif [ "${{ contains(inputs.target_devices, 'level_zero:gpu')  }}" = "true" ]; then
```
Comment (aelovikov-intel) [inline .github/workflows/sycl-linux-run-tests.yml:323]: > @aelovikov-intel is it what you were https://github.com/intel/llvm/pull/14325#discussion_r1691818740 about?

No, we still have duplication here (same as line 309).

Comment (KornevNikita) [reply .github/workflows/sycl-linux-run-tests.yml:323]: What about just merging these two steps into one?

Comment (aelovikov-intel) [reply .github/workflows/sycl-linux-run-tests.yml:323]: That's what I initially suggested:
> IMO, we should drop this step and print that information after we've selected the right file.

Comment (KornevNikita) [reply .github/workflows/sycl-linux-run-tests.yml:313]: lol, a bit of attentivenness:D done in [b8d9777](https://github.com/intel/llvm/pull/14807/commits/b8d9777cdf5b5db4c1a2f0f4dfc708d1027ba1d9)

**Comment by KornevNikita** [comment]
test runs:
GPU: https://github.com/intel/llvm/actions/runs/10159377952
CPU: https://github.com/intel/llvm/actions/runs/10159374251
Fails are expected due to pi-port, I turned them off in another PR, but didn't update this branch.

File: .github/workflows/sycl-linux-run-tests.yml [hunk 2]
```diff
@@ -300,36 +300,28 @@ jobs:
...
             cts_exclude_filter=$PWD/devops/cts_exclude_filter_L0_GPU
           fi
+
+          # List excluded SYCL CTS categories
+          cat $cts_exclude_filter
```
Comment (aelovikov-intel) [inline .github/workflows/sycl-linux-run-tests.yml:318]: Please surround with some extra text explaining what the output is. Grouping like on line 364.

Comment (KornevNikita) [reply .github/workflows/sycl-linux-run-tests.yml:321]: Added [783aac9](https://github.com/intel/llvm/pull/14807/commits/783aac9b5bd310312b6e9ec9804af34c7f09f9cb)
https://github.com/intel/llvm/actions/runs/10179152431/job/28154280857

**Review by aelovikov-intel — APPROVED** [review]

File: .github/workflows/sycl-linux-run-tests.yml [hunk 3]
```diff
@@ -300,36 +300,32 @@ jobs:
...
+          echo "::group::Excluded test categories"
+          cat $cts_exclude_filter
+          echo "::endgroup::"
         fi
 
         cmake -GNinja -B./build-cts -S./khronos_sycl_cts -DCMAKE_CXX_COMPILER=$(which clang++) \
         -DSYCL_IMPLEMENTATION=DPCPP \
         -DSYCL_CTS_EXCLUDE_TEST_CATEGORIES="$cts_exclude_filter" \
         -DSYCL_CTS_ENABLE_OPENCL_INTEROP_TESTS=OFF \
-        -DSYCL_CTS_MEASURE_BUILD_TIMES=ON \
```
Comment (aelovikov-intel) [inline .github/workflows/sycl-linux-run-tests.yml:332]: Why are we dropping this?

Comment (KornevNikita) [reply .github/workflows/sycl-linux-run-tests.yml:332]: Mentioned this in the commit message (hope I'm not wrong)
> Also remove DSYCL_CTS_MEASURE_BUILD_TIMES. It affects the performance, but we don't use these results.

Comment (KornevNikita) [reply .github/workflows/sycl-linux-run-tests.yml:332]: @aelovikov-intel should I remove it?

Comment (aelovikov-intel) [reply .github/workflows/sycl-linux-run-tests.yml:332]: If you think it's better without it, then dropping it is fine. You're the one working with the CTS runs in CI.

Comment (KornevNikita) [reply .github/workflows/sycl-linux-run-tests.yml:332]: Okay, then I'd prefer to drop it

**Comment by KornevNikita** [comment]
AFAIU https://github.com/intel/llvm/pull/14980 should fix the failure.
@intel/llvm-gatekeepers could you please merge

**Comment by steffenlarsen** [comment]
@KornevNikita - Could you please summarize the improvements made in the description?

**Pull request merged**